src/main/java/seedu/address/model/reminder/UniqueReminderList.java

```diff
@@ -0,0 +1,139 @@
+ * to be added or deleted to maintain the uniqueness of the ReminderList
+ * Supports a minimal set of list operations.
+ *
+ */
+public class UniqueReminderList implements Iterable<Reminder> {
+
+    private boolean isReminderListDirty = false;
+
+    private final Model model;
+    private final ObservableList<Reminder> internalList;
+    private final ObservableList<Reminder> internalUnmodifiableList;
+
+    /**
+     * Constructor of UniqueReminderList
+     */
+    public UniqueReminderList(Model model) {
+        this.model = model;
+        internalList = FXCollections.observableArrayList();
+        internalUnmodifiableList = FXCollections.unmodifiableObservableList(internalList);
+    }
+
+    /**
+     * Sets the reminder list to be dirty.
+     * To be called when the reminder list is modified.
+     */
+    public void setReminderListDirty() {
+        isReminderListDirty = true;
+    }
+
+    /**
+     * Returns true if the list contains no Reminders.
+     */
+    public boolean isEmpty() {
+        return internalList.isEmpty();
+    }
+
+    /**
+     * Returns true if the list contains an equivalent Reminder as the given argument.
+     */
+    public boolean contains(Reminder toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Reminder> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns the earliest reminder time in the {@code UniqueReminderList}
+     */
+    public long getEarliestReminderTime() {
+        if (internalList.isEmpty()) {
+            return -1;
+        }
+        return internalList.stream().mapToLong(Reminder::getDueTime).min().getAsLong();
+    }
+
+    /**
+     * Returns the list of reminders associated with a specific date.
+     *
+     * @param date The date that which is used to retrieve the list of reminders.
+     * @return the list of reminders mapped from the given date.
+     */
+    public ObservableList<Reminder> getRemindersAfterDate(LocalDate date) {
+        ObservableList<Reminder> reminderList = FXCollections.observableArrayList();
+        List<Reminder> retrievedReminders = internalList.stream()
+                .filter(a -> a.isAfter(date))
+                .collect(Collectors.toList());
+
+        if (retrievedReminders == null) {
+            return reminderList;
+        }
+        reminderList.addAll(retrievedReminders);
+        return reminderList;
+    }
+
+    /**
+     * Updates the entire internal list of reminders using the list of persons in {@code model}.
+     */
+    public void updateReminders() {
+
+        internalList.clear();
+        for (Person person : model.getAddressBook().getPersonList()) {
+            person.updateReminder();
+            person.getReminder().ifPresent(internalList::add);
+        }
+    }
```

Comment:
I don't think the ReminderList class should be importing model though, better to just pass the personList as an argument?

Reply:
https://github.com/AY2324S1-CS2103T-F11-4/tp/pull/107#discussion_r1377658641

Reply:
Oh ok I gotz